pallets/file-system/src/types.rs

```diff
@@ -0,0 +1,61 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::BoundedVec;
+use frame_system::pallet_prelude::BlockNumberFor;
+use scale_info::TypeInfo;
+
+use crate::Config;
+
+/// Metadata for a storage request.
+///
+/// This is used to track the status of a storage request
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, Debug, PartialEq, Eq, Clone)]
+#[scale_info(skip_type_params(T))]
+pub struct StorageRequestMetadata<T: Config> {
```

Comment:
Shouldn't this also hold the user that started the request? Not just he's multiaddress